### PR TITLE
csi: update cmdline args as used by ceph-csi

### PR DESCRIPTION
**Pull request by iPraveenParihar**
**Description**
Syncing with current cephcsi deployment files 

Added below cmdline args to sidecars
- csi-resizer
   - "--feature-gates=RecoverVolumeExpansionFailure=true"
 - csi-provisioner
    - "--feature-gates=HonorPVReclaimPolicy=true"
     - "--prevent-volume-mode-conversion=true"
 
 
**Links**
- https://github.com/ceph/ceph-csi/pull/3186
- https://github.com/ceph/ceph-csi/pull/3078
- https://github.com/ceph/ceph-csi/pull/4279

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by iPraveenParihar** [comment]
WIP

**Comment by iPraveenParihar** [comment]
> It seems odd to enable feature gates by default. If we really need the feature gates, seems like we should have an option to enable it or not, which is disabled by default.

we have it enabled by default in ceph-csi.

**Comment by travisn** [comment]
@iPraveenParihar Please rebase to pick up latest CI fixes

**Review by travisn — CHANGES_REQUESTED** [review]
Both the helm and smoke test suites are failing, it looks like the PVs are not being deleted. There must be some issue with these flags.
```
2024-01-05 05:12:46.168397 I | integrationTest: Waiting for PV "pvc-93b030f3-d24f-4233-99d2-4c99a76a07ba" to have status "" with exists false
2024-01-05 05:12:51.171807 I | integrationTest: Waiting for PV "pvc-93b030f3-d24f-4233-99d2-4c99a76a07ba" to have status "" with exists false
    ceph_base_block_test.go:345: 
        	Error Trace:	/home/runner/work/rook/rook/tests/integration/ceph_base_block_test.go:345
        	            				/home/runner/work/rook/rook/tests/integration/ceph_base_block_test.go:320
        	            				/home/runner/work/rook/rook/tests/integration/ceph_smoke_test.go:117
        	Error:      	Should be true
        	Test:       	TestCephSmokeSuite/TestBlockStorage_SmokeTest
```

**Review by travisn — COMMENTED** [review]
The CI looks good now, just some known intermittent failures. @Madhu-1 could you also review?

File: pkg/operator/ceph/csi/template/nfs/csi-nfsplugin-provisioner-dep.yaml [hunk 1]
```diff
@@ -93,6 +94,8 @@ spec:
             - "--leader-election-lease-duration=137s"
             - "--leader-election-renew-deadline=107s"
             - "--leader-election-retry-period=26s"
+            - "--feature-gates=HonorPVReclaimPolicy=true"
+            - "--prevent-volume-mode-conversion=true"
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/template/nfs/csi-nfsplugin-provisioner-dep.yaml:98]: this is not required for cephfs and nfs as we don't support blockmode PVC but no harm in adding it as well :)

File: deploy/examples/common.yaml [hunk 2]
```diff
@@ -125,7 +125,7 @@ rules:
     verbs: ["get", "list", "watch"]
   - apiGroups: [""]
     resources: ["persistentvolumes"]
-    verbs: ["get", "list", "watch", "create", "delete", "patch"]
+    verbs: ["get", "list", "watch", "create", "update", "delete", "patch"]
```
Comment (Madhu-1) [inline deploy/examples/common.yaml:128]: why do we need PV update access? i don't see its required for external-provisioner/external-resizer

**Review by Madhu-1 — COMMENTED** [review]
LGTM, left a comment related to RBAC. once clarified we are good to get it merged

File: deploy/examples/common.yaml (hunk 2 above)
Comment (iPraveenParihar) [reply deploy/examples/common.yaml:128]: CI was failing at deleting the PV.
```
E0112 09:10:40.783738       1 controller.go:1007] error syncing volume "pvc-d8b8c06c-49f2-4d71-b866-714164ee11fe": persistentvolumes "pvc-d8b8c06c-49f2-4d71-b866-714164ee11fe" is forbidden: User "system:serviceaccount:smoke-ns-system:rook-csi-rbd-provisioner-sa" cannot update resource "persistentvolumes" in API group "" at the cluster scope
```

we also have it in ceph-csi 
https://github.com/ceph/ceph-csi/blob/200fbdbf28591c404a85888d8bfc9561fa23dbb8/deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml#L25

Comment (Madhu-1) [reply deploy/examples/common.yaml:128]: it might be dependent on the sidecar version. i have not seen this in latest version of sidecar here

**Review by Madhu-1 — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Pull request merged**